ihatemoney/utils.py

```diff
@@ -131,3 +132,38 @@ def list_of_dicts2csv(dict_to_convert):
 
 # base64 encoding that works with both py2 and py3 and yield no warning
 base64_encode = base64.encodestring if six.PY2 else base64.encodebytes
+
+
+class LoginThrottler():
+    """Simple login throttler used to limit authentification attempts based on client's ip address
+    When using multiple workers, remaining number of attempts can get inconsistent
+    but will still be limited to num_workers * max_attempts
+    """
+    def __init__(self, max_attempts=3, delay=1):
+        self._max_attempts = max_attempts
+        # Delay in minutes before attempts counter is reset
+        self._delay = delay
+        self._attempts = {}
+
+    def get_remaining_attempts(self, ip):
+        return self._max_attempts - self._attempts.get(ip, [datetime.now(), 0])[1]
+
+    def increment_attempts_counter(self, ip):
+        if self._attempts.get(ip) is None:
+            # Store first attempt date and number of attempts since
+            self._attempts[ip] = [datetime.now(), 0]
```

Comment:
To fake random IP addresses, a client would manipulate the X-FORWARDED-FOR header. ProxyFix offers a protection against that because by default, if there are multiple proxies, it only trusts the headers set by the last one. In non-proxy setups, it means that the headers set by the client would be trusted and that the OOM killer might pay us a visit.
TLDR: Yeah let's add an auto-reset :)